flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/scheduler/SlotSharingGroup.java

```diff
@@ -70,12 +73,35 @@ public ResourceProfile getResourceProfile() {
         return resourceProfile;
     }
 
+    public String getSlotSharingGroupName() {
+        return slotSharingGroupName;
+    }
+
+    public void setSlotSharingGroupName(String slotSharingGroupName) {
+        this.slotSharingGroupName = slotSharingGroupName;
+    }
+
     // ------------------------------------------------------------------------
     //  Utilities
     // ------------------------------------------------------------------------
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(slotSharingGroupId);
+    }
+
```

Comment:
If we use this class as the map key and only consider the immutable part, I think the equals method is also necessary.